src/resources/dataPoints/multiQueryFetchHelpers.ts

```diff
@@ -0,0 +1,142 @@
+// Copyright 2019 Cognite AS
+
+import _ from 'lodash';
+import ms from 'ms';
+import {
+  DatapointsGetAggregateDatapoint,
+  DatapointsGetDatapoint,
+  DatapointsMultiQuery,
+  ItemsWrapper,
+} from '../..';
+import { API_MAX_DATAPOINTS_PER_REQUEST } from '../../constants';
+import { CDFHttpClient } from '../../utils/http/cdfHttpClient';
+import {
+  ConsolidatedQueryValues,
+  PotentiallyUndefinedQueryValues,
+  QueryResponse,
+} from './types';
+
+function cleanQueryInput(
+  q: DatapointsMultiQuery
+): PotentiallyUndefinedQueryValues {
+  function alwaysReturnNumber(value: string | number | Date): number {
+    if (typeof value === 'string') {
+      return ms(value);
+    }
+    if (value instanceof Date) {
+      return value.getMilliseconds();
+    }
+    return value;
+  }
+  function numberOrUndefined(
+    // tslint:disable-next-line:max-union-size
+    value: number | string | Date | undefined
+  ): number | undefined {
+    return typeof value === 'undefined' ? value : alwaysReturnNumber(value);
+  }
+  const start = numberOrUndefined(q.start);
+  const end = numberOrUndefined(q.end);
+  const limit = numberOrUndefined(q.limit);
+  const granularity: undefined | number = numberOrUndefined(q.granularity);
+
+  return { start, end, limit, granularity };
+}
+
+function extrapolateStart(
+  end: number,
+  granularity: number,
+  limit: number
+): number {
+  return end - granularity * limit;
+}
+
+function extrapolateEnd(
+  start: number,
+  granularity: number,
+  limit: number
+): number {
+  return start + granularity * limit;
+}
+
+function extrapolateLimit(
+  start: number,
+  end: number,
+  granularity: number
+): number {
+  return Math.ceil((end - start) / granularity);
+}
+
+function extrapolateGranularity(start: number, end: number, limit: number) {
+  return Math.floor((start - end) / limit);
+}
+
+export function extrapolateValues(
```

Comment:
these do not make sense, the user should give all values except possible 'no limit', and if there are more than limit datapoints before end, it is cut off at limit.
also note granularity is a string and not a number